src/charm.py

```diff
@@ -28,141 +28,136 @@ class PrometheusScrapeConfigCharm(CharmBase):
     """PrometheusScrapeConfigCharm is a adapter charm.
 
     PrometheusScrapeConfigCharm has no real workload. It
-    transforms incoming scrape jobs from upstreams (e.g.,
-    charms that expose a metrics endpoint) and pushes them
-    to downstream to charms that will execute the scraping
-    (e.g., Prometheus).
+    transforms incoming scrape jobs from metrics providers i.e scrape
+    targets (e.g., charms that expose a metrics endpoint) and pushes them
+    to metrics consumer charms (e.g., a Prometheus charm)
+    that will execute the metrics scraping.
     """
 
     def __init__(self, *args):
-        """Construct the charm, no arguments needed."""
+        """Construct the charm."""
         super().__init__(*args)
 
-        self._target_relation_name = "configurable-scrape-jobs"
-        self._prometheus_relation_name = "metrics-endpoint"
+        self._metrics_provider_relation_name = "configurable-scrape-jobs"
+        self._metrics_consumer_relation_name = "metrics-endpoint"
```

Comment:
Convert to class variables?

Reply:
i.e. "static" variables. If you're so inclined, use ClassVar
https://docs.python.org/3/library/typing.html#typing.ClassVar

Reply:
There is no real benefit to doing this right now. If unit tests need this in the future we can move it else where then. Leaving it in the most private scope possible when there is no specific use case for making it more visible would be the right thing to do.